src/katgpucbf/xbgpu/engine.py

```diff
@@ -860,28 +841,14 @@ def free_rx_item(self, item: RxQueueItem) -> None:
         if item.refcount == 0:
             # All Pipelines are done with this item
             self._active_in_sem.release()
-            # Need to wait for any outstanding events before recycling
-            self.add_service_task(
-                asyncio.create_task(
-                    self._push_recv_chunk(item),
-                    name="Receive Chunk Recycle Task",
-                )
-            )
+            # TODO: Don't recycle the chunk in the reset of the Item itself
+            # Need to separate the concerns
+            item.chunk.recycle()  # type: ignore
```

Comment:
Rather than suppressing mypy, help it out with something like `assert item.chunk is not None`.